src/HealthChecks.Publisher.ApplicationInsights/ApplicationInsightsPublisher.cs

```diff
@@ -48,8 +49,31 @@ public Task PublishAsync(HealthReport report, CancellationToken cancellationToke
                 SaveGeneralizedReport(report, client);
             }
 
+            if (_trackExceptions)
+            {
+                TrackExceptions(report, client);
+            }
+
             return Task.CompletedTask;
         }
+        private void TrackExceptions(HealthReport report, TelemetryClient client)

```

Comment:
If the health report contains exception, you can save also detailed report and exception? probably detailed and generalized report will contain exception information without create any other method

Reply:
That's a good question. Initially, I was going to `TelemetryClient.TrackException` only in the detailed report, but I think it's information you would like to have even in the summary report, that's why I've added the `TrackExceptions` method.

Also, note that for detailed report you could choose to not report healthy results; it probably doesn't make much sense but it could happen that a report is healthy but it also includes an exception and this is why the `TrackExceptions` method is called regardless you choose a summary or detailed report.